lib/mongo/auth.ex

```diff
@@ -39,6 +39,8 @@ defmodule Mongo.Auth do
     if username && password, do: auth ++ [{username, password}], else: auth
   end
 
+  defp mechanism(%{wire_version: version, auth_mechanism: "MONGODB-X509"}) when version >= 3,
```

Comment:
I would rather the `:auth_mechanism` option to be an atom instead of a string.

Reply:
The JS and Python libs use a string, and the official docs kind of point that way too. (Check the command line client connection at the bottom of https://docs.mongodb.com/manual/tutorial/configure-x509-client-authentication/). Java uses an x509 connection subclass. But, you're the author, so fine by me.